diff -dur freewheeling-0.6.4.orig/src/fweelin_browser.h freewheeling-0.6.4/src/fweelin_browser.h
--- freewheeling-0.6.4.orig/src/fweelin_browser.h	2017-04-03 21:19:44.000000000 +0200
+++ freewheeling-0.6.4/src/fweelin_browser.h	2018-11-20 19:22:05.157793326 +0100
@@ -52,7 +52,7 @@
 class BrowserItem {
  public:
   // Initialize a browser item with name n
-  BrowserItem(char *n = 0, char default_name = 1) : 
+  BrowserItem(const char *n = 0, char default_name = 1) : 
     default_name(default_name), next(0), prev(0) {
     if (n == 0)
       name = 0;
@@ -415,7 +415,7 @@
 public:
 
   PatchItem (int id = 0, int bank = 0, int prog = 0, int channel = 0,
-             char *name = 0, char bypasscc = 0, int bypasschannel = -1, float bypasstime1 = 0.0, float bypasstime2 = 10.0) :
+             const char *name = 0, char bypasscc = 0, int bypasschannel = -1, float bypasstime1 = 0.0, float bypasstime2 = 10.0) :
     BrowserItem(name), id(id), bank(bank), prog(prog), channel(channel),
     bypasscc(bypasscc), bypasschannel(bypasschannel), bypasstime1(bypasstime1), bypasstime2(bypasstime2), 
     zones(0), numzones(0) {};
diff -dur freewheeling-0.6.4.orig/src/fweelin_fluidsynth.cc freewheeling-0.6.4/src/fweelin_fluidsynth.cc
--- freewheeling-0.6.4.orig/src/fweelin_fluidsynth.cc	2017-04-03 21:19:44.000000000 +0200
+++ freewheeling-0.6.4/src/fweelin_fluidsynth.cc	2018-11-20 19:21:42.736622350 +0100
@@ -33,12 +33,6 @@
 
 #include "fweelin_fluidsynth.h"
 
-#define fluid_sfont_iteration_start(_sf) (*(_sf)->iteration_start)(_sf)
-#define fluid_sfont_iteration_next(_sf,_pr) (*(_sf)->iteration_next)(_sf,_pr)
-#define fluid_preset_get_name(_preset) (*(_preset)->get_name)(_preset)
-#define fluid_preset_get_banknum(_preset) (*(_preset)->get_banknum)(_preset)
-#define fluid_preset_get_num(_preset) (*(_preset)->get_num)(_preset)
-
 void FluidSynthParam_Int::Send(fluid_settings_t *settings) {
   printf("FLUID: Setting parameter '%s' = '%d'\n",name,val);
   fluid_settings_setint(settings, name, val);
@@ -91,11 +85,11 @@
     double pitches[12];
     for (int i = 0; i < 12; i++)
       pitches[i] = tuning;
-    fluid_synth_create_octave_tuning(synth, 0, 0, "DETUNE", pitches);
+    fluid_synth_activate_octave_tuning(synth, 0, 0, "DETUNE", pitches, true);
 
     // Select tuning
     for (int i = 0; i < MAX_MIDI_CHANNELS; i++) 
-      fluid_synth_select_tuning(synth, i, 0, 0);
+      fluid_synth_activate_tuning(synth, i, 0, 0, true);
   } else 
     printf("FLUID: Using default tuning\n");
 
@@ -160,7 +154,7 @@
 
 // Sets up our internal patch list based on loaded soundfonts
 void FluidSynthProcessor::SetupPatches() {
-  fluid_preset_t preset;
+  fluid_preset_t *preset;
 
   PatchBrowser *br = (PatchBrowser *) app->getBROWSER(B_Patch);
 
@@ -174,12 +168,12 @@
     for (int i = 0; i < sfcnt; i++) {
       fluid_sfont_t *curfont = fluid_synth_get_sfont(synth,i);
       fluid_sfont_iteration_start(curfont);
-      while (fluid_sfont_iteration_next(curfont, &preset))
+      while ((preset = fluid_sfont_iteration_next(curfont)))
         br->AddItem(new PatchItem(fluid_sfont_get_id(curfont),
-                                  fluid_preset_get_banknum(&preset),
-                                  fluid_preset_get_num(&preset),
+                                  fluid_preset_get_banknum(preset),
+                                  fluid_preset_get_num(preset),
                                   fluidchan,
-                                  fluid_preset_get_name(&preset)));
+                                  fluid_preset_get_name(preset)));
 
       if (i+1 < sfcnt) {
         // End of soundfont- put in a divider
