diff -dur -x '*~' freewheeling-4a03065f9d2f520cbd37dd560b4cae5685418153.orig/src/fweelin_config.cc freewheeling-4a03065f9d2f520cbd37dd560b4cae5685418153/src/fweelin_config.cc
--- freewheeling-4a03065f9d2f520cbd37dd560b4cae5685418153.orig/src/fweelin_config.cc	2017-02-24 19:47:23.000000000 +0100
+++ freewheeling-4a03065f9d2f520cbd37dd560b4cae5685418153/src/fweelin_config.cc	2017-03-10 19:12:37.894574426 +0100
@@ -690,7 +690,7 @@
   int str_len = strlen(str_base)+4;
   char basebuf[str_len];
   if (contnum == 0)
-    sprintf(basebuf,str_base);
+    sprintf(basebuf,"%s",str_base);
   else 
     snprintf(basebuf,str_len,"%s%d",str_base,contnum);
 
@@ -1140,7 +1140,7 @@
           int str_len = strlen(str_base)+4;
           char buf[str_len];
           if (contnum == 0)
-            sprintf(buf,str_base);
+            sprintf(buf,"%s",str_base);
           else 
             snprintf(buf,str_len,"%s%d",str_base,contnum);
 
@@ -3284,7 +3284,7 @@
       if (n != 0) {
         nw->filename = new char[xmlStrlen(n)+1];
         strcpy(nw->filename,(char*)n);
-        printf(nw->filename);
+        printf("%s",nw->filename);
         xmlFree(n);
       }
 
